vulcanai2/tests/models/test_cnn.py

```diff
@@ -204,4 +204,14 @@ def test_params_multi_input(self, multi_input_cnn,
             for param, opt_param in zip(test_net1.parameters(),
                                         test_net1.optim.param_groups[0]['params']):
                 assert param is opt_param
-        
\ No newline at end of file
+        
+        # Check the params after saving loaading
+        test_net2.save_model()
+        save_path = test_net2.save_path
+        loaded_test_net2 = BaseNetwork.load_model(load_path=save_path)
+        # Warning: Repeated run of this test will resulting dumping
+        # the saved model into the storage
+        load_params = [torch.allclose(param1, param2)
```

Comment:
I'm not exactly sure what `torch.allclose` does but you should also check that the parameters are equivalent (unless that's what torch.allclose does but then your previous test of fitting 2 separate models and checking allclose with the original model doesn't make too much sense since it's not guaranteed that the weights will be the same after 2 epochs of training)

Reply:
`torch.allclose` does check the parameters element-wise: https://pytorch.org/docs/stable/torch.html#torch.allclose

>checking allclose with the original model doesn't make too much sense since it's not guaranteed that the weights will be the same after 2 epochs of training
Right now the parameters are not the same for when training 2 epochs and the test is passing. Not sure if I quite understood what you mean it does not makes sense

Reply:
aah I see. thanks for the clarification!